### PR TITLE
Profile textfields devas 1439

### PR DESCRIPTION
**Pull request by cedricmessiant**
We only handle text field for now (DEVAS-1439)

### CODE REVIEW

File: assembl/static2/flow/object_types.js [hunk 1]
```diff
@@ -113,3 +113,13 @@ type Extract = {
   extractAction: string,
   body: string
 };
+
+type ConfigurableField = {
```
Comment (cedricmessiant) [inline assembl/static2/flow/object_types.js]: I think we should create a task about flow types. We should probably create multiple files for types relative to different parts of the code

File: assembl/static2/js/app/components/login/signupForm.jsx [hunk 2]
```diff
@@ -106,41 +100,23 @@ class SignupForm extends React.Component<void, SignupFormProps, SignupFormState>
...
-              <FormControl
-                type="password"
-                name="password2"
-                required
-                placeholder={I18n.t('login.password2')}
-                onChange={this.handleInput}
-              />
-            </FormGroup>
-
+            {textFields &&
```
Comment (cedricmessiant) [inline assembl/static2/js/app/components/login/signupForm.jsx]: We should add unit tests for this component

File: assembl/static2/css/components/switchButton.scss [hunk 3]
```diff
@@ -182,3 +182,47 @@ $rsbc-theme-flat-color-disabled-border : darken($rsbc-theme-flat-color-disabled,
     }
   }
 }
+
+// switchButton style for the administration
```
Comment (cedricmessiant) [inline assembl/static2/css/components/switchButton.scss]: too bad we don't have the same style (or something similar) elsewhere...
you can add a story for this one too. Just copy the 3 `SwitchButton` stories that I wrote in `stories/common.jsx` add wrap the switch button in a `div.administration` should do the trick

File: assembl/graphql/configurable_fields.py [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+    value_data = GenericScalar(required=True)
+
+
+class FieldDataInput(graphene.InputObjectType, FieldData):
+    pass
+
+
+class UpdateProfileFields(graphene.Mutation):
+    class Input:
+        data = graphene.List(FieldDataInput, required=True)
```
[the remark this answers is not in the document]
Comment (vincentfretin) [reply assembl/graphql/configurable_fields.py]: yes

File: assembl/alembic/versions/2a7357d60fa5_fix_landing_page_module_types.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+            sa.text(
+                'UPDATE landing_page_module SET "type"=:val WHERE landing_page_module.type=:old_type'
+            ).bindparams(old_type='resource', val='landing_page_module')
+        )
+        mark_changed()
+
+
+def downgrade(pyramid_env):
+    with context.begin_transaction():
+        pass
```
[the remark this answers is not in the document]
Comment (vincentfretin) [reply assembl/alembic/versions/2a7357d60fa5_fix_landing_page_module_types.py]: Yes, this is the case when you switch to another branch @cedricmessiant :) and you need to downgrade.

File: assembl/alembic/versions/4a5c5c46d729_add_profile_text_fields.py [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+                nullable=False, index=False),
+            sa.Column('agent_profile_id',
+                sa.Integer,
+                sa.ForeignKey('agent_profile.id', ondelete="CASCADE", onupdate="CASCADE"),
+                nullable=False, index=False),
+            sa.Column('value_data', JSONB),
+        )
+
+        # insert default text fields
+        with m.TextField.default_db.no_autoflush as db:
```
[the remark this answers is not in the document]
Comment (cedricmessiant) [reply assembl/alembic/versions/4a5c5c46d729_add_profile_text_fields.py]: ok, but if the migration fails and the tables are created you will have to comment all the code above to be able to run the rest of the migration

File: assembl/graphql/configurable_fields.py [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+                profile_field_id = field_info['id']
+                profile_field_id = int(Node.from_global_id(profile_field_id)[1])
+                profile_field = cls.get(profile_field_id)
+                if profile_field:
+                    require_instance_permission(CrudPermissions.UPDATE, profile_field, context)
+                    profile_field.value_data = field_info['value_data']
+                else:
+                    configurable_field_id = field_info['configurable_field_id']
+                    configurable_field_id = int(Node.from_global_id(configurable_field_id)[1])
+                    configurable_field = models.AbstractConfigurableField.get(configurable_field_id)
```
[the remark this answers is not in the document]
Comment (cedricmessiant) [reply assembl/graphql/configurable_fields.py]: @ayyazdaniaryan I'm not sure to understand which code you want me to move and where. Could you give me more details please ? (the 3 lines that I have for context of this comment are not about `AgentProfile`)

File: assembl/alembic/versions/4a5c5c46d729_add_profile_text_fields.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (vincentfretin) [reply assembl/alembic/versions/4a5c5c46d729_add_profile_text_fields.py]: I agree with @cedricmessiant . Our current alembic skeleton is wrong in this regard. The alembic step should only have one transaction, not several. If there are two transactions and the second fail, you have an inconsistent state of the upgrade. And as @cedricmessiant said, you will need to comment all the code of the first transaction to be able to run the rest of the alembic step.

File: assembl/alembic/versions/2a7357d60fa5_fix_landing_page_module_types.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (cedricmessiant) [reply assembl/alembic/versions/2a7357d60fa5_fix_landing_page_module_types.py]: Right, I see !

File: assembl/graphql/configurable_fields.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (cedricmessiant) [reply assembl/graphql/configurable_fields.py]: well, I don't see how to move this on `AgentProfile` model in a clean way (it is really tightly linked to the mutation and all the configurable field stuff)

**Pull request merged**